dateutil/rrule.py

```diff
@@ -678,6 +696,14 @@ def __str__(self):
         dateutil-specific extension BYEASTER.
         """
 
+        def _fmt(obj):
```

Comment:
What is the purpose of this function? The other two branches you added are never hit.

Reply:
Without this function, strings are formated uncorrectly by: `parts.append(partfmt.format(name=name, vals=(','.join(str(v) for v in value))))`. (The second branch never hit because I didn't add tests, but the last one may not be used.)